app/src/main/java/com/rkm/tasky/network/authentication/implementation/AuthenticationManagerImpl.kt

```diff
@@ -54,14 +56,29 @@ class AuthenticationManagerImpl @Inject constructor(
     private fun scheduleWork() {
         val constraints = Constraints.Builder().setRequiredNetworkType(NetworkType.CONNECTED)
             .setRequiresBatteryNotLow(true).build()
-        val periodicSyncWorkRequest = PeriodicWorkRequestBuilder<SyncWorker>(30, TimeUnit.MINUTES)
+        val periodicCreateSyncWorkRequest = PeriodicWorkRequestBuilder<CreateSyncWorker>(30, TimeUnit.MINUTES)
             .setConstraints(constraints)
-            .addTag(SyncWorker.SYNC_WORKER_TAG)
             .build()
+        val periodicDeleteSyncWorkRequest = PeriodicWorkRequestBuilder<DeleteSyncWorker>(30, TimeUnit.MINUTES)
+            .setConstraints(constraints)
+            .build()
+        val periodicUpdateSyncWorkRequest = PeriodicWorkRequestBuilder<UpdateSyncWorker>(30, TimeUnit.MINUTES)
+            .setConstraints(constraints)
+            .build()
+        workManager.enqueueUniquePeriodicWork(
+            CreateSyncWorker.CREATE_SYNC_WORKER_TAG,
+            ExistingPeriodicWorkPolicy.UPDATE,
+            periodicCreateSyncWorkRequest
+        )
+        workManager.enqueueUniquePeriodicWork(
+            DeleteSyncWorker.DELETE_SYNC_WORKER_TAG,
+            ExistingPeriodicWorkPolicy.UPDATE,
+            periodicDeleteSyncWorkRequest
+        )
         workManager.enqueueUniquePeriodicWork(
-            SyncWorker.SYNC_WORKER_TAG,
+            UpdateSyncWorker.UPDATE_SYNC_WORKER_TAG,
             ExistingPeriodicWorkPolicy.UPDATE,
-            periodicSyncWorkRequest
+            periodicUpdateSyncWorkRequest
         )
```

Comment:
Do these all possibly execute in parallel? Wouldn't you need to make sure that a specific order is followed? For example if you create an event in offline mode and then also update it, you would need to sync them in that order with the API as well. Or you somehow merge them into one create event, but it has to be handled in some way